### DISCUSSION TITLE
MacOS bad permission with custom datadir location

### DISCUSSION THREAD

**Discussion by ecker00** [nextcloud/all-in-one#1114] [Questions] [answered]
Running on an M1 MacOS server, and specifying a custom `NEXTCLOUD_DATADIR` and I followed [documentation](https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir) that said to change ownership/permissions on MacOS like:
```sh
sudo chown -R 33:0 /var/nextcloud-data
sudo chmod -R 750 /var/nextcloud-data
```

But on my machine user 33:0 is `_appstore:wheel` and not `www-data:???` which I assume it's trying to set. The `www-data` user doesn't exist.  But by default there is a `_www`  user which I guess have the same purpose? – But it doesn't work either. I always get the following error:

```sh
touch: /mnt/ncdata/this-is-a-test-file: Operation not permitted
The www-data user doesn't seem to have access rights in /mnt/ncdata.
Did you maybe change the datadir and did forget to apply the correct permissions?
```

I guess I need to create the `www-data` user, but under which group?

**Comment by szaimen** [comment]
Hi, you may need to add the directory to shared directories in docker desktop first. See https://stackoverflow.com/a/65602048.

Can you confirm that it works then?

**Comment by ecker00** [comment, marked as answer]
Thank you for the pointer @szaimen, didn't seem to work still. But realised that I don't need to set a custom `NEXTCLOUD_DATADIR` value, which caused this to happen. I was thinking that I needed my own backup solution (so needed easy access to the files) like with most services, but seems the AIO got that built in solution for backups, perfect! 👍
